[Toh Hong Xian] iP

raythx98 opened the pull request.

Commit:
Add Gradle support

Commit:
Fix Gradle support

Gradle does not show texts encoded in "UTF-8" properly.

Added appropriate lines in build.gradle to display encoded texts
properly.

Let's display texts corrected, decoded using "UTF-8" format.

For discussion on how to display "UTF-8" properly on gradle:
nus-cs2103-AY2021S1/forum#123

raythx98 committed:
Level-1

raythx98 committed:
Level-1 improved

raythx98 committed:
Level-2

raythx98 committed:
Level-3

raythx98 committed:
Level-4

raythx98 committed:
Level-5

raythx98 committed:
A-Enums

raythx98 committed:
Level-6.1

raythx98 committed:
A-TextUiTesting

raythx98 committed:
Level-6.2

raythx98 committed:
Level-6.3

raythx98 committed:
Add some changes

Test changes

raythx98 committed:
Level-7

Add save feature

raythx98 committed:
Add date and time format

Date and time is handled as a string.

Added the LocalDate and LocalTime class so Duke is able to parse and evaluate time

raythx98 committed:
Add more OOP

Current implementation floods the Duke class

Let's use more OOP to structure our code better!

raythx98 committed:
Improve OOP

raythx98 committed:
Add package to all classes

raythx98 committed:
Add JUnit test

raythx98 committed:
Add Jar Build

raythx98 committed:
Add javadocs

raythx98 committed:
Change according to Java Coding Standard

raythx98 committed:
add Find feature

raythx98 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/TaskList.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/ToDos.java

raythx98 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/task/Task.java

raythx98 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/ToDos.java

raythx98 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

raythx98 committed:
fix minor bugs

raythx98 committed:
Add missing branch

raythx98 committed:
Add missing branch

raythx98 committed:
Merge branch 'branch-Level-8'

raythx98 committed:
add changes

raythx98 committed:
Add changes

raythx98 committed:
Merge branch 'branch-Level-8'

raythx98 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

raythx98 committed:
Add files to support checkstyle

raythx98 committed:
Add files to support checkstyle

raythx98 committed:
Add varargs to handle multiple arguments

raythx98 committed:
Add varargs to handle multiple arguments

raythx98 committed:
Add changes to improve code quality

raythx98 committed:
update runtest.bat to handle missing directory.

The classpath specified in the .bat file is incorrect, and program does
not handle missing directory appropriately.

Updated the classpath of .bat file and handled missing directories.

Let's update code to handle missing directory, and we will delete these directories everytime we start the .bat file.

Refer to this discussion on handling missing directory:
nus-cs2103-AY2021S1/forum#47
nus-cs2103-AY2021S1/forum#121

raythx98 committed:
update code for more OOP

Current code has too much reuse of code.

Let's update our code to abstract out more methods.

raythx98 committed:
Include JavaFX

Added files to implement the Graphic User Interface for Duke!

Let's move on to GUI from CLI

raythx98 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	build.gradle
#	data/tasks.txt
#	src/main/java/Duke.java
#	src/main/java/duke/Ui.java
#	src/main/java/raythx98/duke/task/Deadline.java
#	src/main/java/raythx98/duke/task/Event.java
#	src/main/java/raythx98/duke/task/ToDos.java

raythx98 committed:
fix bug

ameliatjy commented on src/main/java/raythx98/duke/Duke.java:
```diff
@@ -0,0 +1,45 @@
+        tasks = storage.load();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Perhaps you could edit this Javadoc comment to describe your completed method?

raythx98 replied:
Agreed, will resolve

ameliatjy commented on src/main/java/raythx98/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,14 @@
+
+public abstract class AddCommand extends Command {
+    protected final String taskDescriptionDeadline;
+    public AddCommand(String taskDescriptionDeadline) {
```
Should this public method come along with a Javadoc comment?

ameliatjy commented on src/main/java/raythx98/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,14 @@
+    public AddCommand(String taskDescriptionDeadline) {
+        this.taskDescriptionDeadline = taskDescriptionDeadline;
+    }
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
```
Should this public method come along with a Javadoc comment? I noticed similar issues in other classes as well.

raythx98 replied:
Agreed, will resolve for all cases, thank you for pointing it out!

ameliatjy commented on src/main/java/raythx98/duke/parser/Parser.java:
```diff
@@ -0,0 +1,104 @@
+     *
+     * @param tasks    List of tasks at the moment.
+     * @param nextLine String to be parsed.
+     * @return
```
Is there a missing statement within the Javadoc comment to describe the what the method returns?

raythx98 replied:
Agreed, will resolve

ameliatjy commented on src/main/java/raythx98/duke/ui/Ui.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class Ui {
+
+    private String toShowOnScreen = "";
```
Perhaps the variable name should be a noun instead?

raythx98 replied:
Agreed, will resolve

ameliatjy reviewed (commented):
Chanced by and saw that there were not any PR reviews so thought I'll give some, hopefully they are accurate and will be of help! There were largely many missing Javadoc comments for public methods but code has followed the naming standards well, good job on that!

howtoosee commented on src/main/java/raythx98/duke/Duke.java:
```diff
@@ -0,0 +1,45 @@
+    private Parser parser;
+
+    public Duke(String filePath) {
+        ui = new Ui();
```
consider referencing instance level variables with `this.` to mitigate variable shadowing and be clearer

raythx98 replied:
Look at:
https://se-education.org/guides/conventions/java/index.html
under variables where it states "Use the this keyword only when a field is shadowed by a method or constructor parameter."

Also, please leave only 1 review (with multiple comments) to avoid spamming other contributors' inboxes, thank you :)

howtoosee commented on src/main/java/raythx98/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,14 @@
+import raythx98.duke.task.TaskList;
+import raythx98.duke.ui.Ui;
+
+public abstract class AddCommand extends Command {
```
Missing javadoc for class, similarly for other classes too.

howtoosee commented on src/main/java/raythx98/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,30 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        String[] descriptionSplit = taskDescriptionDeadline.split(" /by ", 2);
+        Task task;
```
Consider naming variables differently to the class name for clarity

howtoosee commented on src/main/java/raythx98/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,30 @@
+        }
+        tasks.addTask(task);
+        ui.addShowOnScreen("Got it, here yur task bij\n" + task.toString()
+                + "\nNow you have " + tasks.getSize() + " tasks in the list.");
```
consider extracting the addShowOnScreen to the parent class AddCommand, or Command itself

hakujitsu commented on src/main/java/raythx98/duke/task/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+import raythx98.duke.exception.InvalidIndexException;
+
+/**
+ * Deals with manipulation on the list of tasks
```
Should this JavaDoc comment end with a full stop?

raythx98 replied:
Agreed, will resolve

howtoosee commented on src/main/java/raythx98/duke/storage/Writer.java:
```diff
@@ -0,0 +1,52 @@
+            File file = new File("data");
+            //Creating the directory
+            boolean bool = file.mkdir();
+            if (bool) {
```
Consider giving this boolean a more meaningful name

howtoosee commented on src/main/java/raythx98/duke/task/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+    protected LocalDate date = null;
+    protected String by = null;
+
+    public Deadline(String description, String by) {
```
variable name "by" is not very intuitive. maybe use "dateString"

hakujitsu commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,56 @@
+
+public class ParserTest {
+
+    private final PrintStream standardOut = System.out;
+    private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
```
Should these constant names be capitalised?

raythx98 replied:
I'm not sure, these constants are not static, does it follow the convention to change them to STANDARD_OUT and OUTPUT_STREAM_CAPTOR?

raythx98 replied:
Yep I checked on this, will resolve

howtoosee commented on src/main/java/raythx98/duke/task/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+        return "Got it, here yur task bij\n" + match.toString();
+    }
+
+    public String toString() {
```
missing `@Override` tag

hakujitsu commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,56 @@
+    @Test
+    public void parse_byeEntry_success() {
+
+        boolean actualResult = new Parser().parse(new TaskList(), "bye");
```
Should this boolean be named to sound like a boolean?

raythx98 replied:
Agreed, will resolve

howtoosee commented on src/main/java/raythx98/duke/task/ToDos.java:
```diff
@@ -0,0 +1,78 @@
+        return tag;
+    }
+
+    public String toPrint() {
```
method name should be a commanding word starting with a verb, such as 'getStringForPrinting'

howtoosee reviewed (commented):
overall code is quite neat with some marginal coding standards not followed properly. good job!

hakujitsu commented on src/main/java/raythx98/duke/task/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    public String findTask(String str) throws InvalidIndexException {
+        TaskList match = new TaskList();
+        for (Task i: taskList) {
```
Should there be a space before the colon?

raythx98 replied:
Agreed, will resolve

hakujitsu commented on src/main/java/raythx98/duke/task/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    public String toString() {
+        String string = "";
```
Should this String have a more descriptive name?

raythx98 replied:
Agreed, will resolve

hakujitsu commented on src/main/java/raythx98/duke/task/ToDos.java:
```diff
@@ -0,0 +1,78 @@
+/**
+ * Represents a specific task which is a to do.
+ */
+public class ToDos extends Task {
```
Should this class be named with a singular noun instead of a plural noun?

raythx98 replied:
Agreed, will resolve

hakujitsu reviewed (commented):
LGTM, your code was very clear and almost always followed the coding standard. Good job 👍

raythx98 commented:
> overall code is quite neat with some marginal coding standards not followed properly. good job!

Thank you for the comments! Will make the appropriate adjustments!

raythx98 commented:
> LGTM, your code was very clear and almost always followed the coding standard. Good job 👍

Thank you for the comments! Will make the appropriate changes!

raythx98 commented:
> Chanced by and saw that there were not any PR reviews so thought I'll give some, hopefully they are accurate and will be of help! There were largely many missing Javadoc comments for public methods but code has followed the naming standards well, good job on that!

Thank you! Will make the appropriate changes!

raythx98 committed:
Add help command

Added help command and other UI improvements, repackaged duke to grandma

Let's rename our application to grandma!

raythx98 committed:
Improve UI, code quality

raythx98 committed:
Add and handle exceptions

Added a myriad of exceptions to support grandma

Let's start practising good exception catching practice!

raythx98 committed:
Add assertions

Added assumptions in various points in the project using assertions

Let's start using assertions toverify assymptions in our code

raythx98 committed:
Improve code quality

Code is not adhering the Java Coding Standard.

Modified the project to adhere to the Java Coding Standard.

Let's start practicing good java coding standard!

raythx98 committed:
Merge pull request #1 from raythx98/branch-A-Assertions

Add assertions

raythx98 committed:
Merge branch 'master' into branch-A-CodeQuality

raythx98 committed:
Merge pull request #2 from raythx98/branch-A-CodeQuality

Improve code quality

raythx98 committed:
Add tag feature

Updated the project to support tagging using #{tag}

raythx98 committed:
Improve the GUI

Minor fixes and subtle changes to the GUI

raythx98 committed:
Update user guide

raythx98 committed:
Update README.md

raythx98 committed:
Add Ui screenshot to user guide

raythx98 committed:
Fix JUnit test

raythx98 committed:
Fix minor bugs

raythx98 committed:
Fix minor bugs

raythx98 committed:
Fix date and time bugs

Previously, the string input for date time format was errornous.

Corrected and program no longer throws an unchecked exception.

raythx98 committed:
Update README.md

raythx98 committed:
Refine find function

Previously, the find function can only search by task description.

Included search through tags.

raythx98 committed:
Update UI to reflect changes in program

raythx98 committed:
Update git ignore files

raythx98 committed:
Set theme jekyll-theme-hacker

raythx98 committed:
Untrack files in .gitignore

raythx98 committed:
Merge branch 'master' of https://github.com/raythx98/ip

raythx98 committed:
Fix minor bugs

raythx98 committed:
Untrack files in .gitignore

raythx98 committed:
Add data to .gitignore

raythx98 committed:
Update README.md

raythx98 committed:
Update README.md

raythx98 committed:
Fix final bugs

raythx98 committed:
Scale README image

raythx98 committed:
Minor Bug Fix

raythx98 committed:
Update README.md

raythx98 committed:
Update README.md

raythx98 committed:
Update Documentation
